src/Select.js

```diff
@@ -177,7 +177,7 @@ var Select = React.createClass({
 					newProps.placeholder)
 				);
 			};
-			if (this.props.asyncOptions) {
+			if (this.props.asyncOptions && this.props.autoload) {
```

Comment:
> So if `this.props.asyncOptions` is specified and `newProps.value` is a string then it should always call `this.props.asyncOptions` to get the full label/value object

But what about string values that appears as labels? Such a values doesn’t need options to show label because the value is the label. So that logic will trigger unnecessary asyncOptions call.

I think we should respect flag `autoload` and never load options automatically before user’s actions.
Consider the grid with lots of react-select components. They all will simultaneously request options on initial value update.